### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by kryvets-vadym**
[DEMO LINK](https://kryvets-vadym.github.io/react_todo-app-with-api/)

### CODE REVIEW

File: src/api/todos.ts [hunk 1]
```diff
@@ -2,7 +2,26 @@ import { Todo } from '../types/Todo';
...
+  const newTodo = await client.post('/todos', todo);
+
+  return newTodo;
+};
+
+export const removeTodo = (todoId: number) => {
+  return client.delete(`/todos/${todoId}`);
+};
+
+// type EditParametr = 'title' | 'completed';
```
Comment (DanilSerzh) [inline src/api/todos.ts:20]: remove comments

File: src/components/TodoItem/TodoItem.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    }
+  });
+
+  const editTodoById = async () => {
+    await editTodo(todo.id, { completed: !todo.completed });
+
+    await loadTodos();
+  };
+
+  // const handleDoubleClick = (event: React.MouseEvent<HTMLSpanElement>) => {
```
Comment (DanilSerzh) [inline src/components/TodoItem/TodoItem.tsx:43]: comments

File: src/components/TodoItem/TodoItem.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+  const onUpdateTitle = async () => {
+    if (todo.title === title) {
+      setIsEditFormVisible(!isEditFormVisible);
+
+      return;
+    }
+
+    try {
+      if (title === '') {
```
Comment (DanilSerzh) [inline src/components/TodoItem/TodoItem.tsx:57]: !title except this

File: src/components/AddFiled/AddField.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  );
+
+  return (
+    <header className="todoapp__header">
+      {isTodosAvailable && (
+        <button
+          data-cy="ToggleAllButton"
+          aria-label="Toggle all"
+          type="button"
+          // className="todoapp__toggle-all"
```
Comment (DanilSerzh) [inline src/components/AddFiled/AddField.tsx:92]: comment

**Review by DanilSerzh — APPROVED** [review]
small corrections here and in the previous ones and it will be great

**Review by Olexandr6 — APPROVED** [review]

**Pull request closed without merging**